gateway-service-impl/src/main/java/org/hypertrace/gateway/service/explore/entity/EntityRequestHandler.java

```diff
@@ -90,15 +94,18 @@ public ExploreResponse.Builder handleRequest(
       requestContext.setHasGroupBy(true);
     }
 
-    Set<String> entityIds = getEntityIds(requestContext, exploreRequest);
     ExploreResponse.Builder builder = ExploreResponse.newBuilder();
-
-    if (entityIds.isEmpty()) {
-      return builder;
+    Set<String> entityIds = new HashSet<>();
+    if (!exploreRequest.getIncludeNonLiveEntities()) {
```

Comment:
A bit confused. Where are we checking for fetching results time agnostically?

Reply:
because say there are only non-live entities. then if we don't put this check results won't be returned. Secondly, if non live entities are selected filter and group by wont work properly because we are putting entity ids filter which are in live traffic only